src/Akka.Persistence.Azure/Journal/AzureTableStorageJournal.cs

```diff
@@ -761,16 +755,17 @@ private async Task<long> ReplayTaggedMessagesAsync(ReplayTaggedMessages replay,
                         counter++;
                     }
 
-                    maxOrderingId = Math.Max(maxOrderingId, entry.UtcTicks);
+                    currentOrderingId = Math.Max(currentOrderingId, entry.UtcTicks);
+                    
+                    if (counter >= replay.Max)
+                        return new ReplayTaggedMessageSuccess(false);
                 }
 
                 if (counter >= replay.Max)
-                {
-                    break;
-                }
+                    return new ReplayTaggedMessageSuccess(false);
             }
 
-            return maxOrderingId;
+            return new ReplayTaggedMessageSuccess(true);
```

Comment:
Records returned by all page is less than the maximum requested, we have reached the last row from the query.